### PR TITLE
Bluetooth: Controller: Fix CIS encryption when DF support enabled

### PR DESCRIPTION
**Pull request by cvinayak**
Fix hung Controller when establishing CIS on an encrypted ACL connection with Controller built with direction finding support enabled.

### CODE REVIEW

**Review by jhedberg — APPROVED** [review]

File: subsys/bluetooth/controller/ll_sw/nordic/hal/nrf5/radio/radio.c [hunk 1]
```diff
@@ -1692,20 +1692,25 @@ static void *radio_ccm_ext_rx_pkt_set(struct ccm *cnf, uint8_t phy, uint8_t pdu_
...
+		} else if (pdu_type == RADIO_PKT_CONF_PDU_TYPE_DC) {
+			/* When direction finding CTE receive feature is enabled then on-the-fly
+			 * PDU parsing for CTEInfo is always done. In such situation, the CCM
+			 * TASKS_CRYPT must be started with short delay. That give the Radio time
+			 * to store received bits in shared memory.
+			 */
+			radio_bc_configure(CCM_TASKS_CRYPT_DELAY_BITS);
+			radio_bc_status_reset();
+			hal_trigger_crypt_by_bcmatch_ppi_config();
+			hal_radio_nrf_ppi_channels_enable(BIT(HAL_TRIGGER_CRYPT_DELAY_PPI));
```
Comment (Thalley) [inline subsys/bluetooth/controller/ll_sw/nordic/hal/nrf5/radio/radio.c:1708]: Hmm, are any of these unavailable when `CONFIG_BT_CTLR_DF_CONN_CTE_RX=n`? 

I am just wondering if the `#if defined(CONFIG_BT_CTLR_DF_CONN_CTE_RX)` could be replaced by a `if (IS_ENABLED)` which would make this look better

Comment (cvinayak) [reply subsys/bluetooth/controller/ll_sw/nordic/hal/nrf5/radio/radio.c:1708]: `CCM_TASKS_CRYPT_DELAY_BITS` and `HAL_TRIGGER_CRYPT_DELAY_PPI` both are only defined under `CONFIG_BT_CTLR_DF_CONN_CTE_RX`

Comment (Thalley) [reply subsys/bluetooth/controller/ll_sw/nordic/hal/nrf5/radio/radio.c:1708]: Ah. What purpose is there to guard those defines?

**Review by Thalley — APPROVED** [review]

File: subsys/bluetooth/controller/ll_sw/nordic/hal/nrf5/radio/radio.c (hunk 1 above)
Comment (cvinayak) [reply subsys/bluetooth/controller/ll_sw/nordic/hal/nrf5/radio/radio.c:1708]: `HAL_TRIGGER_CRYPT_DELAY_PPI`  PPI instance/resource would then either be available or unavailable for application use:

https://github.com/zephyrproject-rtos/zephyr/blob/4d598683972a3601c2e90b53d58ffbb6e33e99ea/subsys/bluetooth/controller/ll_sw/nordic/hal/nrf5/nrfx_glue/bt_ctlr_used_resources.h#L108-L113

Comment (Thalley) [reply subsys/bluetooth/controller/ll_sw/nordic/hal/nrf5/radio/radio.c:1708]: Is that a problem? 

At least in the host we rarely (if ever) guard `#define`s, as they do no harm.

Comment (cvinayak) [reply subsys/bluetooth/controller/ll_sw/nordic/hal/nrf5/radio/radio.c:1708]: I have not checked if that is a problem (does not seem so), but the defines (header file using) are used externally (nrfx module) and will need more change/testing.

**Pull request merged**